# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1401916563 14400
#      Wed Jun 04 17:16:03 2014 -0400
# Node ID ebf6a1e5ff59cebc600104ea6203ce7f18a5c3ba
# Parent  307a67b0425383ba94eb19b5250b69a37f176920
load files in private directories from private functions (bug #42500)

* load-path.cc (find_private_file): Don't skip search when current
function is private.

diff --git a/libinterp/corefcn/load-path.cc b/libinterp/corefcn/load-path.cc
--- a/libinterp/corefcn/load-path.cc
+++ b/libinterp/corefcn/load-path.cc
@@ -1262,18 +1262,22 @@ find_private_file (const std::string& fn
 {
   std::string retval;
 
   // Look in private directory corresponding to current function (if
   // any).
 
   octave_user_function *curr_fcn = symbol_table::get_curr_fcn ();
 
-  if (curr_fcn && ! curr_fcn->is_private_function ())
+  if (curr_fcn)
     {
+      // Even for private functions, dir_name doesn't contain the
+      // "private" directory component so we append it here in all
+      // cases.
+
       std::string dir_name = curr_fcn->dir_name ();
 
       if (! dir_name.empty ())
         {
           std::string pfname = dir_name + file_ops::dir_sep_str ()
             + "private" + file_ops::dir_sep_str () + fname;
 
           file_stat fs (pfname);
